src/Raven.Server/ServerWide/ClusterStateMachine.License.cs

```diff
@@ -391,72 +393,149 @@ void AssertAutoIndexesCount()
+            subscriptionsNamesPerDatabase[command.DatabaseName].Add(command.SubscriptionName);
         }
 
-        long GetSubscriptionsCount()
+        if (AssertNumberOfSubscriptionsPerClusterLimits(serverStore, items, context, subscriptionsNamesPerDatabase))
+            return putSubscriptionCommandsList;
+
+        AssertNumberOfSubscriptionsPerDatabaseLimits(serverStore, items, context, subscriptionsNamesPerDatabase);
+        return putSubscriptionCommandsList;
+    }
+
+    private bool AssertNumberOfSubscriptionsPerDatabaseLimits(
+        ServerStore serverStore,
+        Table items,
+        ClusterOperationContext context,
+        IReadOnlyDictionary<string, List<string>> subscriptionsNamesPerDatabase)
+    {
+        foreach ((string databaseName, List<string> subscriptionsNames) in subscriptionsNamesPerDatabase)
         {
-            long count = 0;
+            var subscriptionsCount = GetSubscriptionsCountForDatabase(context.Allocator, items, databaseName, subscriptionsNames);
 
-            foreach (var name in GetDatabaseNames(context))
-            {
-                count += GetSubscriptionsCountForDatabase(context.Allocator, items, name);
-            }
+            var maxSubscriptionsPerDatabase = serverStore.LicenseManager.LicenseStatus.MaxNumberOfSubscriptionsPerDatabase;
+            if (maxSubscriptionsPerDatabase is not >= 0)
+                return false;
 
-            return count;
+            if (subscriptionsCount + subscriptionsNames.Count > maxSubscriptionsPerDatabase == false)
+                return false;
+
+            if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion60000) == false)
+                return true;
+
+            throw new LicenseLimitException(LimitType.Subscriptions,
+                $"The maximum number of subscriptions per database cannot exceed the limit of: {maxSubscriptionsPerDatabase}");
         }
+
+        return false;
+    }
+
+    private bool AssertNumberOfSubscriptionsPerClusterLimits(
+        ServerStore serverStore,
+        Table items,
+        ClusterOperationContext context,
+        IReadOnlyDictionary<string, List<string>> subscriptionsNamesPerDatabase)
+    {
+        var maxSubscriptionsPerCluster = serverStore.LicenseManager.LicenseStatus.MaxNumberOfSubscriptionsPerCluster;
+        if (maxSubscriptionsPerCluster is not >= 0)
+            return false;
+
+        var clusterSubscriptionsCounts =
+            GetDatabaseNames(context)
+                .Sum(databaseName => GetSubscriptionsCountForDatabase(context.Allocator, items, databaseName, subscriptionsNamesPerDatabase[databaseName]));
+
+        var subscriptionCommandsCount = subscriptionsNamesPerDatabase.Sum(x => x.Value.Count);
+        if (clusterSubscriptionsCounts + subscriptionCommandsCount > maxSubscriptionsPerCluster == false)
+            return false;
+
+        if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion60000) == false)
```

Comment:
check this at the beginning?

Reply:
I suggest the following order:
1. Check if there's a licensing limitation.
2. Check if CanAssertLicenseLimits().
3. Perform the check.

I've done it this way for now. If there are objections, I'm ready to consider them.